memory/jemalloc_nodump_allocator.cc

```diff
@@ -105,45 +113,71 @@ void JemallocNodumpAllocator::Deallocate(void* p) {
   dallocx(p, tcache_flag);
 }
 
-Status JemallocNodumpAllocator::InitializeArenas() {
-  // Create arena.
-  size_t arena_index_size = sizeof(arena_index_);
-  int ret =
-      mallctl("arenas.create", &arena_index_, &arena_index_size, nullptr, 0);
-  if (ret != 0) {
-    return Status::Incomplete("Failed to create jemalloc arena, error code: " +
-                              std::to_string(ret));
+uint32_t JemallocNodumpAllocator::GetArenaIndex() const {
+  if (arena_indexes_.size() == 1) {
+    return arena_indexes_[0];
   }
-  assert(arena_index_ != 0);
 
-  // Read existing hooks.
-  std::string key = "arena." + std::to_string(arena_index_) + ".extent_hooks";
-  extent_hooks_t* hooks;
-  size_t hooks_size = sizeof(hooks);
-  ret = mallctl(key.c_str(), &hooks, &hooks_size, nullptr, 0);
-  if (ret != 0) {
-    return Status::Incomplete("Failed to read existing hooks, error code: " +
-                              std::to_string(ret));
-  }
+  static std::atomic<uint32_t> next_seed = 0;
```

Comment:
Late feedback, but `next_seed` should have started at `1`, because the seed gets preprocessed and `0` becomes `1`:

https://github.com/facebook/rocksdb/blob/41a7fbf7583670bf084144529bb31314771197ff/util/random.h#LL34C1-L34C1

So if we start at `0` the first two threads are equivalently seeded. Should not be a big issue because they'll likely quickly lose sync, but worth addressing if you touch this code again.